pipeline/matching/core/analytics.py

```diff
@@ -0,0 +1,27 @@
+from pipeline.matching.core import MatchingEngine
+from pipeline.types import MatchingInput, MatchingInternalAnalytics
+
+
+class CollectInternalAnalytics:
+    def __init__(self, name, matching_engine):
+        self.name: str = name
+        self.matching_engine: MatchingEngine = matching_engine
```

Comment:
You don't need either of these properties, this can be an empty constructor.

```python
class CollectInternalAnalytics:
    def __init__(self):
        pass
```